### DISCUSSION TITLE
Delivery selection calculation and multishipping logic is sometimes wrong

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#37665] [Ideas]
### The problem

If there is no carrier that can send all products in the cart, prestashop will create a multishipping/multicarrier delivery option that consists of more carriers.

But, this can be sometimes bad if you have a personal pickup carrier that can "send" everything.

#### Imagine you have following carriers
- DHL
- Fedex
- Personal pickup at Paris

#### And you have following products
- Product A can be sent with DHL and Personal pickup at Paris.
- Product B can be sent with Fedex and Personal pickup at Paris.

#### You add them both to cart , you go to checkout and you will see only the personal pickup, so people who are not local cannot get it.
- Personal pickup at Paris

#### But, you should ideally see this:
- DHL + Fedex (multishipping)
- Personal pickup at Paris

#### Because if Personal pickup at Paris wasn't enabled, you would actually see the option:
- DHL + Fedex (multishipping)

### How to resolve this? 🤔 

Maybe - add a "Carrier type" radio with options "Classic/Personal pickup" to carrier options? Then Prestashop will calculate multishipping for these two groups separately.

It could also resolve this ticket - https://github.com/PrestaShop/PrestaShop/discussions/37664.

### Additional context

Ping @MateoCastilloPS @ibahloul-ps @kpodemski 

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by MateoCastilloPS** [comment]
Hi @Hlavtox adding the notion of carrier type into prestashop is something we are currently discussing internally, it would definitely improve the way carriers can be filtered / displayed in the checkout, and not only for this use case but for many others !  @Nakahiru what do you think ?
